### PR TITLE
Transform camera component

### PR DESCRIPTION
**Pull request by jasonjk192**

### CODE REVIEW

File: Include/ComponentComposition/CameraComponent.h [hunk 1]
```diff
@@ -4,20 +4,60 @@
...
 #include "Maths/Quaternion.h"
 #include "ComponentComposition/Component.h"
+#include "ComponentComposition/Color.h"
+#include "ComponentComposition/GameObject.h"
+#include "ComponentComposition/TransformComponent.h"
 
 namespace PlatinumEngine
 {
 	enum projectionType {PERSPECTIVE, ORTHOGRAPHIC};
+	enum clearFlagType {SKYBOX, BGCOLOR, DEPTH, NONE};
```
Comment (ShawnWicky) [inline Include/ComponentComposition/CameraComponent.h:14]: Why we have SKYBOX mode in camera component? Maybe that is the responsibilty of sceneCamera? I'm not sure about that.

Comment (kablouser) [reply Include/ComponentComposition/CameraComponent.h:14]: I think it's fine. But the enum name should be capital, use enum class instead and lowercase for the values.

Comment (jasonjk192) [reply Include/ComponentComposition/CameraComponent.h]: Whoops...forgot to commit the changes earlier but they are up now

File: Include/ComponentComposition/TransformComponent.h [hunk 2]
```diff
@@ -4,28 +4,67 @@
...
+		void Rotate(Maths::Quaternion q, relativeTo space);
+		void Rotate(Maths::Vec3 axis, float angle, relativeTo space);
 		void RotateAround(Maths::Vec3 point, Maths::Vec3 axis, float angle);
-		void Scale(Maths::Vec3 scale);
 		void LookAt(Maths::Vec3 target);
+
+		Maths::Vec3 TransformDirection(Maths::Vec3 dir);
+		Maths::Vec3 InverseTransformDirection(Maths::Vec3 dir);
+		Maths::Vec3 TransformPoint(Maths::Vec3 point);
+		Maths::Vec3 InverseTransformPoint(Maths::Vec3 point);
```
Comment (kablouser) [inline Include/ComponentComposition/TransformComponent.h:41]: A lot of support. Very complicated.

File: Include/ComponentComposition/TransformComponent.h [hunk 3]
```diff
@@ -4,28 +4,67 @@
...
+
+		Maths::Vec3 GetLocalPosition();
+		Maths::Quaternion GetWorldRotation();
+
+		Maths::Vec3 forward();
+		void forward(Maths::Vec3 v);
+		Maths::Vec3 up();
+		void up(Maths::Vec3 v);
+		Maths::Vec3 right();
+		void right(Maths::Vec3 v);
```
Comment (kablouser) [inline Include/ComponentComposition/TransformComponent.h:55]: You should name them differently i.e. get/set

**Review by kablouser — COMMENTED** [review]
Generally, much more functions than we need. And when you write more functions, more tech debt, more bugs, longer to implement. Start simple, then if we really need extra helper functions, add them later.

Don't copy code from a 6 years old unofficial project https://github.com/tezheng/UH5
If you need guidance ask the group. I can give you my old transform code, or I can help you write the header file/API.

**Pull request merged**